### DISCUSSION TITLE
DateEntry ValueError

### DISCUSSION THREAD

**Discussion by filipE321** [israel-dryer/ttkbootstrap#466] [Q&A] [answered]
I have noticed that if you enter a wierd format like for example "20203-06-" or "2023-06" you get this error: 
raise ValueError("time data %r does not match format %r" %
ValueError: time data '2023-06' does not match format '%x'

I figured out that this most likely comes from the fact that when you write something in the input field and then click the calendar button, the widget tries to make that input into a date so that it can jump to it in the calendar. I have looked around but I cant seem to find a solution. 

Any ideas? Please help.

**Comment by windir19** [comment]
I hope you do find an answer. It seems like there are a lot of problems with the ttkbootstrap implementation of DateEntry. I have my own that is different than yours. I am wondering if the OG developer is not working on this library anymore. Very sad since overall I like it.

**Comment by Megsman** [comment]
The entry box in the DateEntry takes a date object.

**Comment by israel-dryer** [comment, marked as answer]
**Fixed** (available since **v1.18.0**). This was an unguarded date parse in the calendar-button handler (`_on_date_ask`): when you typed free-form text and clicked the calendar, the widget did `datetime.strptime(text, format)` with no error handling, so malformed input like `2023-06` raised `ValueError`. Good diagnosis, @filipE321 — that's exactly where it came from.

**On current 1.x** the parse is guarded. Unparseable entry text no longer crashes; it emits a warning and the calendar-button click is simply a no-op (the popup doesn't open). If you *want* the strict old behavior, opt in at construction:

```python
ttk.DateEntry(root, raise_exception=True)   # default is False -> warn instead of raise
```

**In the upcoming 2.0** this is handled more gracefully still: the entry text is parsed tolerantly (configured format -> a few common formats -> ISO-8601), so several inputs that used to fail now parse. When the text genuinely can't be parsed, the widget falls back to the last-set date and **still opens the calendar** instead of dead-ending — and `raise_exception=True` remains available for strict callers.

Closing this out as resolved. Thanks for the clear repro.
